lib/rucio/core/did_meta_plugins/elasticsearch_meta.py

```diff
@@ -0,0 +1,406 @@
+                    query["search_after"] = search_after
+                    query.pop("from", None)
+                # Execute search
+                results = self.client.search(body=query)
+                hits = results['hits']['hits']
+                if not hits:
+                    break
+
+                for hit in hits:
+                    did_full = f"{hit['_source']['scope']}:{hit['_source']['name']}"
+                    if did_full not in ignore_dids:
+                        ignore_dids.append(did_full)
+                        if long:
+                            yield {
+                                'scope': (hit['_source']['scope']),
+                                'name': hit['_source']['name'],
+                                'did_type': hit['_source'].get('did_type', 'N/A'),
+                                'bytes': hit['_source'].get('bytes', 'N/A'),
+                                'length': hit['_source'].get('length', 'N/A')
+                            }
+                        else:
+                            yield hit['_source']['name']
+
+                    total_processed += 1
+                    if limit and total_processed >= limit:
+                        break
+
+                # Update search_after for the next iteration
+                search_after = hits[-1]["sort"]
+
+        finally:
+            # Always delete the point in time when done
+            self.client.close_point_in_time(body={"id": pit_id})
+
+        if recursive:
+            raise exception.UnsupportedOperation(f"'{self.plugin_name.lower()}' metadata module does not currently support recursive searches")
+
+    def on_delete(
+        self,
+        scope: "InternalScope",
+        name: str,
+        archive: bool = False,
+        session: "Optional[Session]" = None
+    ) -> None:
+        """
+        Delete a document and optionally archive it.
+
+        :param scope: The scope of the document
+        :param name: The name of the document
+        :param archive: Whether to archive the document before deletion
+        :raises DataIdentifierNotFound: If the DID is not found.
+        :raises RucioException: If an error occurs while setting the metadata.
+        """
+        doc_id = f"{scope}{name}"
+
+        try:
+            doc = self.client.get(index=self.index, id=doc_id)
+
+            if archive:
+                archived_doc = doc['_source']
+                archived_doc['deleted_at'] = datetime.datetime.now(datetime.timezone.utc).isoformat()
+                self.client.index(index=self.archive_index, id=doc_id, body=archived_doc)
+
+            self.client.delete(index=self.index, id=doc_id)
+
+        except elastic_exceptions.NotFoundError as err:
+            raise exception.DataIdentifierNotFound(f"No metadata found for DID '{scope}:{name}' not found") from err
+        except Exception as err:
+            raise exception.RucioException(err)
+
+    def get_metadata_archived(
```

Comment:
I think this would be better:

```suggestion
    def get_archived_metadata(
```

Reply:
@rdimaio ,
The code for get_metadata_archived and on_delete are added to provide the feature that is in PR : https://github.com/rucio/rucio/pull/6727 . Thus I kept the same name.